### PR TITLE
92 use monolog logging

### PR DESCRIPTION
**Pull request by UlrichKu**
| Q             | A
| ------------- | ---
| Branch        | master
| Bug fix?      | no
| New feature?  | no
| BC breaks?    | yes
| Deprecations? | yes
| Tests pass?   | yes
| Fixed issues  | chameleon-system/chameleon-system#92
| License       | MIT

### CODE REVIEW

File: src/ShopOrderStatusBundle/Resources/config/services.xml [hunk 1]
```diff
@@ -0,0 +1,6 @@
+<?xml version="1.0" ?>
+<container xmlns="http://symfony.com/schema/dic/services"
```
[the remark this answers is not in the document]
Comment (UlrichKu) [reply src/ShopOrderStatusBundle/Resources/config/services.xml]: the extension class is new

File: src/AmazonPaymentBundle/DependencyInjection/ChameleonSystemAmazonPaymentExtension.php [hunk 2]
```diff
@@ -36,4 +37,12 @@ public function load(array $config, ContainerBuilder $container)
         $serviceDefDataFetcher = $container->getDefinition('chameleon_system_amazon_payment.config_provider');
         $serviceDefDataFetcher->replaceArgument(0, $config);
     }
+
+    /**
+     * Allow an extension to prepend the extension configurations.
+     */
+    public function prepend(ContainerBuilder $container)
+    {
+        $container->prependExtensionConfig('monolog', ['channels' => ['order_amazon']]);
```
[the remark this answers is not in the document]
Comment (UlrichKu) [reply src/AmazonPaymentBundle/DependencyInjection/ChameleonSystemAmazonPaymentExtension.php]: Using order_payment_amazon now.

This is the only channel of its kind though.
I didn't really find much for "IPN". So there is no channel for that.

File: src/ShopBundle/objects/TCMSSmartURLHandler/TCMSSmartURLHandler_ShopPayPalIPN.class.php [hunk 3]
```diff
@@ -9,43 +9,42 @@
...
+use Psr\Log\LoggerInterface;
 
 class TCMSSmartURLHandler_ShopPayPalIPN extends TCMSSmartURLHandler
 {
     public function GetPageDef()
     {
+        /**
+         * @var $logger LoggerInterface
+         */
+        $logger = ServiceLocator::get('monolog.logger.order_ipn');
```
[the remark this answers is not in the document]
Comment (UlrichKu) [reply src/ShopBundle/objects/TCMSSmartURLHandler/TCMSSmartURLHandler_ShopPayPalIPN.class.php]: ok

File: src/ShopBundle/objects/db/TShopPaymentHandler/TShopPaymentHandlerPayPal_PayViaLink.class.php [hunk 4]
```diff
@@ -125,6 +130,11 @@ protected function GetInstantPaymentNotificationListenerURL($oOrder)
      */
     public function ProcessIPNRequest($oOrder, $aURLParameter)
     {
+        /**
+         * @var $logger LoggerInterface
+         */
+        $logger = ServiceLocator::get('monolog.logger.order');
```
[the remark this answers is not in the document]
Comment (UlrichKu) [reply src/ShopBundle/objects/db/TShopPaymentHandler/TShopPaymentHandlerPayPal_PayViaLink.class.php]: ok

File: src/ShopOrderStatusBundle/objects/TPkgShopOrderStatusManagerEndPoint.class.php [hunk 5]
```diff
@@ -9,23 +9,34 @@
...
      * @var IPkgCmsCoreLog
+     *
+     * @deprecated since 6.3.0
      */
     private $logger = null;
 
+    /**
+     * @param IPkgCmsCoreLog $logger
+     *
+     * @deprecated since 6.3.0 - is not supported anymore; use only getShopLogger() or do your own logging
```
[the remark this answers is not in the document]
Comment (UlrichKu) [reply src/ShopOrderStatusBundle/objects/TPkgShopOrderStatusManagerEndPoint.class.php]: The method is public?

So somebody might be able to call it.

Or can you clarify what problem you see?

[the remark this answers is not in the document]
Comment (umulmrum) [reply src/ShopOrderStatusBundle/objects/TPkgShopOrderStatusManagerEndPoint.class.php]: Really bad wording in my comment, and I think I can answer it myself. To make it short: Looks good.

**Comment by umulmrum** [comment]
Thanks @UlrichKu

**Pull request merged**